packages/databases-collections/src/components/create-namespace-modal.tsx

```diff
@@ -65,7 +65,14 @@ class CreateDatabaseModal extends PureComponent<
   };
 
   onConfirm = () => {
-    this.props.createNamespace(this.state.data as CreateNamespaceOptions);
+    const isCreateCollection = this.props.databaseName !== null;
+    this.props.createNamespace({
+      ...this.state.data,
+      database: isCreateCollection
+        ? this.state.data.database
+        : this.state.data.database?.trim(),
+      collection: this.state.data.collection?.trim(),
+    } as CreateNamespaceOptions);
```

Comment:
This is application business logic and so belongs in the action, not in the view